### ISSUE TITLE
Profiling tools: pbt2ptt does not handle alignment

### ISSUE THREAD

**Issue by omor1** [ICLDisco/parsec#361]
## Describe the bug
Python's `struct` module does not handle alignment at the end of a structure. Given the following info type and converter code:
```C
typedef struct {
    uint64_t a;
    uint32_t b;
} example_info_t;
static const char *example_conv = "a{uint64_t};b{uint32_t}"
static int example_sk, example_ek;
...
static void example_prof_init(void)
{
    parsec_profiling_add_dictionary_keyword("EXAMPLE", "fill:#FF0000",
                                            sizeof(example_info_t), example_conv,
                                            &example_sk, & example_ek);
}
```
`sizeof(example_info_t)` is 16, because the structure is aligned to its largest member; however, the conversion code is `'QI'` with a size of 12. This causes an error when converting to HDF5/PTT.

### Expected behavior
The conversion code should keep track of the largest data member and add a zero-repeat copy to the end of the conversion code, as described in the [Python documentation](https://docs.python.org/3/library/struct.html). For the example above, this should be `'QI0Q'`.
Alternatively, require users to ensure data member alignment in the info structures and/or conversion codes (bleh).

## Environment (please complete the following information):
 - PaRSEC version: up-to-date profiling subsystem
 - OS: Rocky Linux 8.5
 - Compiler: AOCC 3.0 (LLVM/clang 12.0.0), Python 3.9.4
 - MPI version: N/A

## Additional context
Was trying to add tracking of activations packed into an activation message for the `MPI_ACTIVATE` profiling keyword—this is something I discussed with @bosilca. `parsec_profile_remote_dep_mpi_info_t` is aligned to 8 bytes, since it has a `uint64_t`, but I was only adding 4 bytes for the pack count (and could probably have gotten away with 1 or 2—the maximum number of activations that _can_ be packed is circa 100). For now my workaround will be to just use a 64-bit count instead—it'll have little to no overhead anyway.

**Comment by bosilca** [comment]
The alignment of the struct only matters on the C side, and only before it is passed to the profiling. From that point on it is seen as a contiguous blob of data, the alignement should not matter (unless you have 2 of such structures one after the other). On the side of the python module we read the profiling as a packed stream so I do think it is correct to read 12 bytes for such a struct.

**Comment by omor1** [comment]
```Python
if event_len != len(self):                                                 
    c_string = self.fmt                                                    
    logger.warning('Event %s discarded: expected length differs from provided length (%d != %d)\n'
                   'Check the conversion format <%s>\n',                   
                   event_name, len(self), event_len, c_string)             
    event_len = event_len if event_len < len(self) else len(self)
```
These lines of code causes issues, though I can't tell if this _actually_ discards the event as it claims it does

**Comment by bosilca** [comment]
```diff

diff --git a/tools/profiling/python/pbt2ptt.pyx b/tools/profiling/python/pbt2ptt.pyx
index 4e95296..b7f6b5e 100644
--- a/tools/profiling/python/pbt2ptt.pyx
+++ b/tools/profiling/python/pbt2ptt.pyx
@@ -802,9 +802,15 @@ cdef class ExtendedEvent:
 
         if event_len != len(self):
             c_string = self.fmt
-            logger.warning('Event %s discarded: expected length differs from provided length (%d != %d)\n'
-                           'Check the conversion format <%s>\n',
-                           event_name, len(self), event_len, c_string)
+            if event_len < len(self):
+                logger.warning('Event %s: expected length greater that the reported length (%d != %d)\n'
+                               'The output might get corrupted. Check the conversion format <%s>\n',
+                               event_name, len(self), event_len, c_string)
+            else:
+                logger.info('Event %s padded: expected length differs from the reported length (%d != %d)\n'
+                            'Check the conversion format <%s>\n',
+                            event_name, len(self), event_len, c_string)
+                self.fmt += b"{}x".format(event_len - len(self))
             event_len = event_len if event_len < len(self) else len(self)
         self.event_len = event_len
     def __len__(self):
```

**Comment by bosilca** [comment]
@omor1 here is a possible patch (might be partially correct I dont have access to cython right now).

Your analysis was partially correct, the original code complained about the mismatch in size between what was saved in the profiling file and the size reported by the python struct package, but, unlike the comment suggested, it did not drop the event but instead corrected his size to the one provided by the C struct. My change here will add the missing padding to the python struct, such that it works via the package itself.

I also tested with the struct package that type alignment is correct handled, especially when it generates gaps in the middle of the info structure.

**Comment by omor1** [comment]
I believe this was closed via 03f3f265e47476b57b8fb1fe4ea97a738b3b740d in #363.
